### PR TITLE
Make integrations toggle-able during build

### PR DESCRIPTION
**Pull request by sainak**
## Proposed Changes

>[!NOTE]
>New required environment variables `REACT_SENTRY_DSN`, `REACT_SENTRY_ENABLED`, `REACT_SENTRY_DSN`

- Make integrations toggle-able during build

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Release Notes

- **New Features**
	- Introduced a new `Plausible` integration that currently does not render any UI.
	- Added functions for Sentry integration: `initSentry` and `captureException` for streamlined error handling.

- **Improvements**
	- Enhanced Sentry configuration, including environment variable management.
	- Improved error handling in notifications and hooks by utilizing a dedicated `captureException` method.

- **Bug Fixes**
	- Removed unnecessary Sentry integration from the app component to simplify the rendering logic.

- **Documentation**
	- Updated environment variable definitions to include optional properties for Sentry and Plausible configurations.

### CODE REVIEW

File: care.config.ts [hunk 1]
```diff
@@ -90,10 +90,8 @@ const careConfig = {
   },
 
   sentry: {
-    dsn:
-      env.REACT_SENTRY_DSN ||
-      "https://8801155bd0b848a09de9ebf6f387ebc8@sentry.io/5183632",
-    environment: env.REACT_SENTRY_ENVIRONMENT || "staging",
+    dsn: env.REACT_SENTRY_DSN,
+    environment: env.REACT_SENTRY_ENVIRONMENT || "unknown",
```
[the remark this answers is not in the document]
Comment (sainak) [reply care.config.ts:94]: care.config is inaccessible in vite.config

**Comment by sainak** [comment]
@coderabbitai review

File: src/Integrations/Sentry.tsx [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
-  }, [disabled]);
-
-  return null;
+/**
+ * Captures an exception and sends it to Sentry for monitoring and logging.
+ *
+ * @param {Error | unknown} error - The error object that needs to be captured and sent to Sentry.
+ */
+export function captureException(error: Error | unknown) {
+  Sentry.captureException(error);
```
[the remark this answers is not in the document]
Comment (sainak) [reply src/Integrations/Sentry.tsx:25]: this is not required as the module is replaced with a dummy module which has skeleton handlers based on the main file

File: src/Integrations/Sentry.tsx [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
+      `Sentry configuration incomplete: ${[
+        !careConfig.sentry.dsn && "DSN",
+        !careConfig.sentry.environment && "environment",
+      ]
+        .filter(Boolean)
+        .join(", ")} missing`,
+    );
+    return;
+  }
+  Sentry.init(careConfig.sentry);
```
[the remark this answers is not in the document]
Comment (sainak) [reply src/Integrations/Sentry.tsx:16]: this is not required as the module is replaced with a dummy module which has skeleton handlers based on the main file

**Comment by nihal467** [comment]
@sainak closing this PR due to inactivity, feel free to reopen them once they are ready for review

**Pull request closed without merging**